core/frontend-devtools/src/tools/MapLayerTool.ts

```diff
@@ -54,6 +55,41 @@ class AttachMapLayerBaseTool extends Tool {
     });
   }
 }
+
+/** Attach a map layer from URL base class. */
+export class AttachModelMapLayerTool extends Tool {
+  public static override get minArgs() { return 0; }
+  public static override get maxArgs() { return 1; }
+  public static override toolId = "AttachModelMapLayerTool";
+  constructor(protected _formatId: string) { super(); }
+
+  public override async run(nameIn?: string): Promise<boolean> {
+    const vp = IModelApp.viewManager.selectedView;
+    if (!vp)
+      return false;
+
+    const iModel = vp.iModel;
+    const elements = await iModel.elements.getProps(iModel.selectionSet.elements);
```

Comment:
So this tool will add a maplayer for each model that currently contain a selected elements?

Reply:
Yes.... This is just for testing purposes and I discarded the idea of multiple models for a single layer as too difficult to implement and too complex to understand.